bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -118,4 +134,31 @@ def run_plan(scope, plan_name, named_args = {})
       result
     end
   end
+
+  # Recursively examine the type_expr to build a list of types
+  def extract_parameter_types(type_expr, acc = [])
```

Comment:
This is a little weird since it's both using an accumulator _and_ calling `map`. Do you expect the recursive calls to return their values and be aggregated by the caller or do you expect them to add to `acc`?

I think that if the `acc << value` lines were replaced by `value` and the `map` calls were replaced by `flat_map`, there wouldn't be a need for an accumulator at all.

Reply:
Wow, yeah that makes way more sense. Thanks, this method just became more and more complex I think I got so focused on making it correct that I forgot about making it readable. Udpated to not use acc.